src/cljs/nnts2/note/events.cljs

```diff
@@ -1,20 +1,38 @@
 (ns nnts2.note.events
   (:require [re-frame.core :as re-frame]
-            [nnts2.note.api :as api]))
+            [nnts2.note.api-data :as api-data]))
+
 
 (enable-console-print!)
 
 (re-frame/reg-event-fx
- :note-submit
- (fn [_ [_ note]]
-   (api/create-note note)))
+ ::note-panel-loaded
+ (fn [cofx event]
+   {:db  (:db cofx)
+    :dispatch [::note-get-list]}))
 
-(re-frame/reg-event-db
- :note-submit-success
+
+
+(re-frame/reg-event-fx
+ ::note-submit
+ (fn [cofx [_ note]]
+   {:db (assoc (:db cofx) :note-form note)
+    :http-xhrio (api-data/create-note note)}))
+
+
+(re-frame/reg-event-fx
+ ::note-submit-success
+ (fn [cofx event]
+   {:db  (assoc (:db cofx) :note-form {:title "" :content ""})
+    :dispatch [::note-get-list]}))
```

Comment:
We're firing an event to get the list of notes, but that layer of indirection is not valuable, I think. It's simpler to have a `:http-xhrio` coeffect directly rather than firing off an event.

Reply:
If you think about it, it doesn't really make sense for events to trigger other events...there's really only one event that happened (in this case, the submission of the note).